### PR TITLE
doc(readme): Developers should know about our stack overflow account

### LINKED ISSUE #332: [Feat]: Stack overflow for teams
### Is your feature request related to a problem? Please describe.

We now have a new resource to asking questions at BWTC: https://stackoverflow.com/c/bwtc/questions 

### Describe the solution you'd like.

We should point our developers to this resource in our homepage. Preferably within the "Need Help" Section: https://github.com/Shift3/standards-and-practices#need-help

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by michaelachrisco**
## Changes
1. Modified "Need Help" section within the S&P repo to have links to resources.
2. Our specific stack overflow instance is now referenced within the "Need Help" section.

## Purpose
One of the pain points we have as an organization is not archiving questions and discussions within slack. 

## Approach
Stack Overflow has traditionally been a resource developers go to for longstanding and specific issues.

Closes #332

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,9 +111,11 @@ Using Chocolatey has several advantages:
 Getting stuck is a natural part of development and happens to the best of us. It's better to reach out than to stay silent, and we encourage you to do so. If you find yourself stuck on a task for more than 45 minutes, here are a couple of places to get help:
 
 ##### Slack channels:
-1. **bwtc-technical-discussions** - Great place for technical questions that involve some code.
-2. **Your team channel** - When you start a project, you will be added to a private team channel. Feel free to ask your questions there! Your team lead may be able to help you through your blocker.
 
+
+1. **Your team channel** - When you start a project, you will be added to a private team channel. Feel free to ask your questions there! Your team lead may be able to help you through your blocker.
+2. **[bwtc-stackoverflow](https://stackoverflow.com/c/bwtc/questions)** - Did you know we have our own Stack Overflow? Take a look, it has some great quesions and answers.
```
Comment (apethoud) [inline README.md:117]: Looks like this link implies a Slack channel called `bwtc-stackoverflow`, but I couldn't find one. The link links to our actual hub on StackOverflow's website. If that's the primary location to engage in discussion, should this be documented outside of the "Slack Channels" section of this readme?

Comment (michaelachrisco) [reply README.md:117]: Totally agree, this should not be an implied slack channel. Im thinking of changing "Slack Channels" to "Communication Channels" to make this more broad.

File: README.md [hunk 2]
```diff
@@ -111,9 +111,11 @@ Using Chocolatey has several advantages:
 Getting stuck is a natural part of development and happens to the best of us. It's better to reach out than to stay silent, and we encourage you to do so. If you find yourself stuck on a task for more than 45 minutes, here are a couple of places to get help:
 
 ##### Slack channels:
```
Comment (michaelachrisco) [inline README.md:113]:
```suggestion
##### Communication Channels:
```

**Review by apethoud — APPROVED** [review]
Looks great!

**Pull request merged**